src/components/Commits/CommitsListView.tsx

```diff
@@ -37,17 +43,45 @@ const CommitsListView: React.FC<CommitsListViewProps> = ({
 }) => {
   const navigate = useNavigate();
   const [nameFilter, setNameFilter] = useSearchParam('name', '');
+  const [statusFilterExpanded, setStatusFilterExpanded] = React.useState<boolean>(false);
+  const [statusFiltersParam, setStatusFiltersParam] = useSearchParam('status', '');
+
+  const statusFilters = React.useMemo(
+    () => (statusFiltersParam ? statusFiltersParam.split(',') : []),
+    [statusFiltersParam],
+  );
+
+  const setStatusFilters = React.useCallback(
+    (filters: string[]) => setStatusFiltersParam(filters.join(',')),
+    [setStatusFiltersParam],
+  );
+
+  const statusFilterObj = React.useMemo(() => {
+    return commits.reduce((acc, c) => {
+      const stat = pipelineRunFilterReducer(c.pipelineRuns[0]);
+      if (statuses.includes(stat)) {
+        if (acc[stat] !== undefined) {
+          acc[stat] = acc[stat] + 1;
+        } else {
+          acc[stat] = 1;
+        }
+      }
+      return acc;
+    }, {} as { [key: string]: number });
```

Comment:
Instead of having a Type assertion here, you could add the type to `statusFilterObj` variable.
```suggestion
    }, {});
```